public/app/core/utils/UserProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+
+export interface UserAPI {
+  changePassword: (ChangePassword: ChangePasswordFields) => void;
+}
+
+export interface ChangePasswordFields {
+  oldPassword: string;
+  newPassword: string;
+  confirmNew: string;
+}
+
+export interface Props {
+  children: (api: UserAPI) => JSX.Element;
+}
+
+export interface State {}
+
+export class UserProvider extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {};
+  }
+
+  changePassword = async (payload: ChangePasswordFields) => {
+    await getBackendSrv().put('/api/user/password', payload);
+  };
+
+  render() {
+    const { children } = this.props;
+
+    const api = {
+      changePassword: this.changePassword.bind(this),
```

Comment:
You don't need binding here, do you? `changePassword` is already bound by using `=>` function